src/main/java/fi/aalto/cs/apluscourses/model/APlusAuthentication.java

```diff
@@ -1,33 +1,68 @@
 package fi.aalto.cs.apluscourses.model;
 
+import com.intellij.credentialStore.CredentialAttributes;
+import com.intellij.credentialStore.CredentialAttributesKt;
+import com.intellij.credentialStore.Credentials;
+import com.intellij.ide.passwordSafe.PasswordSafe;
 import java.util.Arrays;
 import org.apache.http.HttpRequest;
 import org.jetbrains.annotations.NotNull;
 import org.jetbrains.annotations.Nullable;
 
 public class APlusAuthentication implements Authentication {
+
+  public static final String AUTHORIZATION_HEADER = "Authorization";
+  public static final String A_COURSES_PLUGIN = "A+ Courses Plugin";
+  public static final String A_API = "A+ API";
+
+  private final Object authenticationLock = new Object();
```

Comment:
I'd remove all `synchronized` blocks from this class because as far as I understand, they are no longer needed.